src/test_mode/src/wrapper.rs

```diff
@@ -41,60 +63,159 @@ pub fn wrapper_init(){
         let jsbsim_systems_path = std::ffi::CString::new("systems").unwrap();
         spresult = fdm_set_systems_path(fdm, jsbsim_systems_path.as_ptr());
 
+        //verify directory structure was set properly
+        if apresult == false || epresult == false || spresult == false {
+            panic!("unable to configure directory structure");
+        }
+        
         //load script prep
-        let script_name = std::ffi::CString::new("run.xml").unwrap();
+        let script_name = std::ffi::CString::new("/scripts/testleds.xml").unwrap();
         let delta_t: f64 = 0.0;
         let init_file = std::ffi::CString::new("").unwrap();
         let lsresult: bool;
 
         //load script
         lsresult = fdm_load_script(fdm, script_name.as_ptr(), delta_t, init_file.as_ptr());
+        
+        //verify script load
+        if lsresult == false{
+            panic!("unable to load script");
+        }
 
-        //run initial
+        //run initial conditions
         let icresult: bool;
         icresult = fdm_run_ic(fdm);
+        
+        //verify initial conditions
+        if icresult == false{
+            panic!("unable to set initial conditions");
+        }
     }
 }
 
-//sends data to jsbsim.  iterates fdm.  gets response from jsbsim.
-//development:  update state variable
-pub fn send_to_jsbsim(newcw: u8, newccw: u8)->bool{
-    let runresult: bool;
+
+///////////////////////////////////////////////////////////////////////////////
+//
+//  Function name:      send_to_jsbsim
+//
+//  Purpose:            update FDM with data from the controller interface
+//
+//  Methodology:
+//  -update fdm property testmode/ledcw with controller output cw
+//  -update fdm property testmode/ledccw with controller output ccw
+//  -this will indicate to the fdm that the flight controller is engaging
+//  -the controller interface i.e. firing thrusters
+//                        
+///////////////////////////////////////////////////////////////////////////////
+pub fn send_to_jsbsim(cw: u8, ccw: u8){
+
     unsafe{
+       
+        //set cw & ccw values in jsbsim
+        let property_cw = std::ffi::CString::new("testmode/ledcw").unwrap();
```

Comment:
It is preferable to use CString::into_raw() rather than as_ptr() since as_ptr() can loose ownership of the memory while it is being used from the C code. I was trying to do this earlier and found that this is necessary on when passing a C-string to and external FFI interface in Windows, but will sometimes work on Linux. See https://doc.rust-lang.org/std/ffi/struct.CString.html#method.into_raw